Userland/Libraries/LibJS/Runtime/DatePrototype.cpp

```diff
@@ -125,22 +141,48 @@ JS_DEFINE_NATIVE_FUNCTION(DatePrototype::set_full_year)
     auto* this_object = typed_this(vm, global_object);
     if (!this_object)
         return {};
-    auto new_year = vm.argument(0).to_i32(global_object);
-    if (vm.exception())
-        return {};
-    // FIXME: Support specifying new month and new day as well.
     auto& datetime = this_object->datetime();
-    auto new_month = datetime.month();
-    auto new_day = datetime.day();
-    datetime.set_time(new_year, new_month, new_day, datetime.hour(), datetime.minute(), datetime.second());
-    return Value { this_object->time() };
+
+    auto arg_or = [&vm, &global_object](size_t i, i32 fallback) { return vm.argument_count() > i ? vm.argument(i).to_i32(global_object) : fallback; };
+
+    auto new_year = arg_or(0, 1970);
```

Comment:
The default value 1970 is not part of the spec :thinking:

https://tc39.es/ecma262/#sec-date.prototype.setfullyear